### PR TITLE
Field Extensions: basic theorems

### PR DESCRIPTION
**Pull request by tirix**
Definition and basic theorems about field extensions, finite field extensions, and degrees of field extensions.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -469149,6 +469232,208 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+    df-fldext $a |- /FldExt = { <. e , f >. | ( ( e e. Field /\ f e. Field )
+      /\ ( f = ( e |`s ( Base ` f ) ) /\ ( Base ` f ) e. ( SubRing ` e ) ) ) }
+      $.
+
+    $( Definition of the field extension degree operation.  (Contributed by
+       Thierry Arnoux, 29-Jul-2023.) $)
+    df-extdg $a |- [:] = ( e e. _V , f e. ( /FldExt " { e } )
+      |-> ( dim ` ( ( subringAlg ` e ) ` ( Base ` f ) ) ) ) $.
+
+    $( Definition of the finite extension relation.  (Contributed by Thierry
```
Comment (avekens) [inline set.mm:469268]: "field" is missing between "finite" and "extension"

File: set.mm [hunk 2]
```diff
@@ -469149,6 +469232,208 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+    $d E e f $.  $d F e f $.
+    $( The field extension is a relation.  (Contributed by Thierry Arnoux,
+       29-Jul-2023.) $)
+    relfldext $p |- Rel /FldExt $=
+      ( ve vf cv cfield wcel wa cbs cress co csubrg cfldext df-fldext relopabiv
+      cfv wceq ) ACZDEBCZDEFQPQGNZHIORPJNEFFABKABLM $.
+
+    $( The field extension relation explicited.  (Contributed by Thierry
+       Arnoux, 29-Jul-2023.) $)
+    brfldext $p |- ( ( E e. Field /\ F e. Field ) -> ( E /FldExt F <-> ( F = (
```
Comment (avekens) [inline set.mm:469291]: What does the prefix "br" mean? Binary Rrelation? I would propose to call this theorem simply "fldext".

File: set.mm [hunk 3]
```diff
@@ -469149,6 +469232,208 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+    $( Field extension implies a structure restriction relation.  (Contributed
+       by Thierry Arnoux, 29-Jul-2023.) $)
+    fldextress $p |- ( E /FldExt F -> F = ( E |`s ( Base ` F ) ) ) $=
+      ( cfldext wbr cbs cfv cress wceq csubrg wcel cfield fldextfld1 fldextfld2
+      co wa wb brfldext syl2anc ibi simpld ) ABCDZBABEFZGNHZUBAIFJZUAUCUDOZUAAK
+      JBKJUAUEPABLABMABQRST $.
+
+    $( The finite extension relation explicited.  (Contributed by Thierry
+       Arnoux, 29-Jul-2023.) $)
+    brfinext $p |- ( E /FldExt F -> ( E /FinExt F <-> ( E [:] F ) e. NN0 ) ) $=
```
Comment (avekens) [inline set.mm:469344]: see remark(s) above

File: set.mm (hunk 1 above)
Comment (tirix) [reply set.mm:469268]: Yes, thanks!

File: set.mm (hunk 2 above)
Comment (tirix) [reply set.mm:469291]: Yes, it stands for binary relation. I wanted to follow the same model as for example:
- `df-lmic` -> [`brlmic`](https://us.metamath.org/mpeuni/brlmic.html)
- `df-ric` -> [`brric`](https://us.metamath.org/mpeuni/brric.html)
- `df-gic` -> [`brgic`](https://us.metamath.org/mpeuni/brgic.html)
- `df-ssc` -> [`brssc`](https://us.metamath.org/mpeuni/brssc.html)
- `df-wdom` -> [`brwdom`](https://us.metamath.org/mpeuni/brwdom.html)
- `df-en` -> [`bren`](https://us.metamath.org/mpeuni/bren.html)
- `df-dom` -> [`brdom`](https://us.metamath.org/mpeuni/brdom.html)
- `df-tpos` -> [`brtpos`](https://us.metamath.org/mpeuni/brtpos.html)
- `df-pss` -> [`brtpos`](https://us.metamath.org/mpeuni/brtpos.html)

Comment (tirix) [reply set.mm:469291]: I'll first push the change for the other comment, and wait for your opinion for the rest.

Comment (avekens) [reply set.mm:469291]: OK, this seems to be a kind of convention I was not aware of.

File: set.mm (hunk 3 above)
Comment (avekens) [reply set.mm:469344]: still "field" missing?

File: set.mm [hunk 4]
```diff
@@ -660665,10 +660950,6 @@ not even needed (it can be any class).  (Contributed by Glauco
       $.
   $}
 
-  $( Plus infinity is not a real number.  (Contributed by Glauco Siliprandi,
-     23-Oct-2021.) $)
-  pnfnre2 $p |- -. +oo e. RR $=
-    ( cpnf cr pnfnre neli ) ABCD $.
```
Comment (avekens) [inline set.mm:660671]: This movement to main should be documented in the file changes-set.txt.

File: set.mm (hunk 3 above)
Comment (tirix) [reply set.mm:469344]: Yes! Added.

File: set.mm (hunk 4 above)
Comment (tirix) [reply set.mm:660671]: Yes! I had prepared the line in `changes-set.txt`, but for some reason I did not include it in the PR.

**Review by avekens — APPROVED** [review]

**Pull request merged**